code/_onclick/hud/screen_objects/text_objects.dm

```diff
@@ -8,3 +8,130 @@
 	screen_loc = "CENTER-7,CENTER-7"
 	maptext_height = 480
 	maptext_width = 480
+
+#define MAX_TIMERS 4
+
+/atom/movable/screen/text/screen_timer
+	var/maptext_style_left = "<span class='maptext' style=font-size:16pt;text-align:center; align='top'>"
+	var/maptext_style_right = "</span>"
+	var/maptext_string
+	var/timer_id
+	var/list/timer_mobs = list()
+
+/atom/movable/screen/text/screen_timer/Initialize(
+		mapload,
+		mobs,
+		timer,
+		text,
+		style_start,
+		style_end,
+		offset_x = 12,
+		offset_y = -70
+	)
+	. = ..()
+
+	if(!islist(mobs))
+		if(!mobs)
+			return
+		mobs = list(mobs)
+	if(!timer || !length(mobs))
+		qdel(src)
+		CRASH("Invalid timer for screen nuke timer!")
+	if(style_start)
+		maptext_style_left = style_start
+	if(style_end)
+		maptext_style_right = style_end
+	maptext_string = text
+	timer_id = timer
+	maptext_x = offset_x
+	maptext_y = offset_y
+	update_maptext()
+	apply_to(mobs)
+	START_PROCESSING(SSprocessing, src)
+
+/atom/movable/screen/text/screen_timer/process()
+	if(!timeleft(timer_id))
+		delete_self()
+		return
+	update_maptext()
+
+/atom/movable/screen/text/screen_timer/proc/apply_to(mobs)
+	if(!islist(mobs))
+		if(!mobs)
+			return
+		mobs = list(mobs)
+	for(var/mob/player in mobs)
+		if(player in timer_mobs)
+			continue
+		
+		player.client?.screen += src
+		timer_mobs += player
+		RegisterSignal(player, COMSIG_MOB_LOGIN, PROC_REF(re_attach))
+		RegisterSignal(player, list(COMSIG_MOB_GHOST, COMSIG_MOB_LOGOUT), PROC_REF(remove_from))
+
+// /atom/movable/screen/text/screen_timer/proc/handle_offset()
+// 	var/existing_timers = 0
```

Comment:
commented out code

Reply:
Was hoping to figure out a way to automatically offset it, not possible now since there's only one instance of this